### PR TITLE
Upgrade to Bazel 5

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

Make the repository use Bazel 5. Bazel 4 and below only work with the deprecated Python 2, which is no longer included in some operating systems such as the latest Mac OS X.

## What are the changes implemented in this PR?

1. Upgrade Bazel to 5.1.1.
2. Upgrade all `@vaticle` dependencies (repositories and artifacts)
3. Update BDD steps as a consequence of upgrading to the latest version of `@vaticle_typedb_behaviour`
4. Update test runners according to the latest version of TypeDB Cluster
5. Update all CI jobs to use Python 3.7

### CODE REVIEW

File: .grabl/automation.yml [hunk 1]
```diff
@@ -40,10 +40,11 @@ build:
       image: vaticle-ubuntu-21.04
       type: foreground
       command: |
-        pyenv global 3.6.10
+        pyenv install 3.7.12
```
Comment (lolski) [inline .grabl/automation.yml:43]: Update all jobs to use Python 3.7.12

File: tools/cluster_test_rule.bzl [hunk 2]
```diff
@@ -35,41 +35,57 @@ def _rule_implementation(ctx):
     # TODO: This code is, mostly, copied from our TypeDB behave test
     cmd = "set -e && TYPEDB_ARCHIVE=%s" % typedb_cluster_distro
     cmd += """
-
-           if test -d typedb_distribution; then
+            function server_start() {
```
Comment (lolski) [inline tools/cluster_test_rule.bzl:38]: Update TypeDB Cluster bootup command according to the latest syntax.

File: tests/integration/test_cluster_failover.py [hunk 3]
```diff
@@ -42,6 +42,26 @@ def setUp(self):
                 client.databases().get("typedb").delete()
             client.databases().create("typedb")
 
+    @staticmethod
+    def server_start(index):
```
Comment (lolski) [inline tests/integration/test_cluster_failover.py:46]: Update TypeDB Cluster bootup command according to the latest syntax.

File: tests/behaviour/concept/type/thingtype/thing_type_steps.py [hunk 4]
```diff
@@ -67,6 +67,20 @@ def step_impl(context: Context, root_label: RootLabel, type_label: str, get_labe
     assert_that(context.get_thing_type(root_label, type_label).as_remote(context.tx()).get_label().name(), is_(get_label))
 
 
+@step("{root_label:RootLabel}({type_label}) set abstract: {is_abstract}; throws exception")
+def step_impl(context: Context, root_label: RootLabel, type_label: str, is_abstract):
```
Comment (lolski) [inline tests/behaviour/concept/type/thingtype/thing_type_steps.py:71]: As a consequence of updating `@vaticle_typedb_behaviour` to the latest version, we need to implement this newly defined step.

File: tests/behaviour/typeql/typeql_steps.py [hunk 5]
```diff
@@ -471,8 +471,8 @@ def step_impl(context: Context):
         assert_that(list(context.tx().query().match(query)), has_length(1))
 
 
-@step("each answer does not satisfy")
+@step("templated typeql match; throws exception")
```
Comment (lolski) [inline tests/behaviour/typeql/typeql_steps.py:474]: Update an existing step, again as a consequence of updating `@vaticle_typedb_behaviour` to the latest version.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**